src/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs

```diff
@@ -224,42 +245,147 @@ public void DirectoryInfo_Create_NullDirectorySecurity()
             DirectoryInfo info = new DirectoryInfo("path");
             if (PlatformDetection.IsFullFramework)
             {
-                Assert.Throws<ArgumentNullException>(() => FileSystemAclExtensions.Create(info, null));
+                AssertExtensions.Throws<ArgumentNullException>(() =>
+                    FileSystemAclExtensions.Create(info, null),
+                    "Value cannot be null.\r\nParameter name: directorySecurity");
             }
             else
             {
-                Assert.Throws<ArgumentNullException>(() => info.Create(null));
+                AssertExtensions.Throws<ArgumentNullException>(() =>
+                    info.Create(null),
+                    "Value cannot be null. (Parameter 'directorySecurity')");
             }
         }
 
         [Fact]
         public void DirectoryInfo_Create_NotFound()
         {
-            DirectoryInfo info = new DirectoryInfo(@"W:\\I\\Do\\Not\\Exist");
+            DirectoryInfo info = new DirectoryInfo(@"W:\I\Do\Not\Exist");
```

Comment:
Better to do this in the temp path with guids for the subdir names to make sure you're testing what you think you're testing. You can be confident that the drive exists and isn't some removable or offline storage. Otherwise you might hit other types of errors.

Reply:
Ok no problem